src/IonEventStream.ts

```diff
@@ -206,17 +230,19 @@ export class IonEventStream {
             if(actualEvent.eventType == IonEventType.STREAM_END && expectedEvent.eventType == IonEventType.STREAM_END) {
                 break;
             } else if(actualEvent.eventType == IonEventType.STREAM_END || expectedEvent.eventType == IonEventType.STREAM_END) {
-                    throw new Error("Different number of comparison sets.");
+                throw new Error("Different number of comparison sets.");
             } else if (!(actualEvent.ionType == IonTypes.LIST || actualEvent.ionType == IonTypes.SEXP)
                 || !(expectedEvent.ionType == IonTypes.LIST || expectedEvent.ionType == IonTypes.SEXP)) {
                 throw new Error("Comparison sets must be lists or s-expressions.");
-            } else if((actualEvent.annotations[0] === 'embedded_documents') as any ^ (expectedEvent.annotations[0] === 'embedded_documents') as any) {
+            } else if(this.isEmbedded(actualEvent) as any ^ this.isEmbedded(expectedEvent) as any) {
                 throw new Error("Both streams should be embedded streams.");
             }
 
+            // both containers has type any as it maybe IonEventStream[] or IonEvent[]
+            // depending on the type compare method will be called
```

Comment:
Since both of these are pretty limited in scope, could we use the `IonEventStream[] | IonEvent[]` type union here? If this adds a lot of complexity, feel free to say no.

Reply:
Yah. Actually I tried to do the Union type but it would add complexity to the method. So I haven't added union type here.